server/instance_cloud_oauth.go

```diff
@@ -99,6 +99,11 @@ func (ci *cloudOAuthInstance) getClientForConnection(connection *Connection) (*j
 	oauth2Conf := ci.GetOAuthConfig()
 	ctx := context.Background()
 	tokenSource := oauth2Conf.TokenSource(ctx, connection.OAuth2Token)
+	if ci.JWTInstance != nil && tokenSource == nil {
```

Comment:
What are the benefits of using `tokenSource == nil` here instead of `connection.OAuth2Token == nil`?

Reply:
@mickmister There are no particular benefits here but we are using `connection.OAuth2Token` now to avoid some extra computations of refreshing the token.